### DISCUSSION TITLE
teleport Puppet ruby template for mongodb

### DISCUSSION THREAD

**Discussion by willsc** [gravitational/teleport#8347] [Q&A]
is there an example puppet ruby template for mongodb atlas connectivity that someone could take me through. The database connectivity template has a reference to an aws key which is incompatible with the what we're trying to do with mongodb in terms of connectivity any pointers would be useful .

Example: I would like something similar but for mongodb atlas

db_service:
  enabled: <%= scope.lookupvar('teleport::db_enable') %>
<% unless scope.lookupvar('teleport::databases').empty? -%>
  databases:
<% scope.lookupvar('teleport::databases').flatten.each do |database| -%>

  - name: <%= database['name'] %>
    description: <%= database['description'] %>
    protocol: <%= database['protocol'] %>
    uri: <%= database['uri'] %>
    aws:  <-------- This is a problem when trying the same thing for mongdb.
  <% database['aws'].sort.each do |key,value| -%>
    <%= key %>: <%= value %>
  <% end -%>
  static_labels:
  <% database['static_labels'].sort.each do |key,value| -%>
    <%= key %>: <%= value %>
  <% end -%>

**Comment by webvictim** [comment]
I would consider changing your template to look whether `database.aws` is set or not, and not write the `aws:` section if it's unset.
